### PR TITLE
userContext issue done

### LINKED ISSUE #97: Create a UserContext
Create a `UserContext` based on `React.createContext` that will provide information about the logged-in user and methods for login and logout.

Use the following article for reference: [How to use React Context effectively](https://kentcdodds.com/blog/how-to-use-react-context-effectively).

Context should contain the following:
- `token` a token for the logged-in user in form of a string, or null if user is not logged in
- `login`: function which sets token to the passed value. It also saves it in local storage.
- `logout`: a function that clears token. It removes value from local storage.

When the application runs (on startup) a token should be loaded from local storage - in case there is nothing there, the user is not logged in.

> *Note*: This is an advanced issue and is here merely in case you want to test your skills. Mentors will provide this functionality when the time comes for it.

### PR DESCRIPTION
**Pull request by MaherHussain**
# Description

PR is about userContext that save data local storage when user log in and remove the data from  local storage when the user log out , it made by Maher and Nuha .

Fixes #97

# How to test?

we have created a Router in App.js called "/usertest" and all work fine ?

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

File: src/client/context/userContext.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  const login = () => {
+    console.log('login');
+    const Apitoken = 'aaa';
+    localStorage.setItem('tokenName', Apitoken);
+    setToken(Apitoken);
+  };
+  const logout = () => {
+    localStorage.removeItem('tokenName');
+    setToken(null);
+    /* alert("token will delete") */
```
Comment (bytewiz) [inline src/client/context/userContext.js:23]: you can remove this commented out line

File: src/client/context/userContext.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  const [token, setToken] = useState(null);
+  useEffect(() => {
+    const tokenFromLocalStorage = localStorage.getItem('tokenName');
+
+    if (tokenFromLocalStorage) {
+      setToken(tokenFromLocalStorage);
+    }
+  }, []);
+  const login = () => {
+    console.log('login');
```
Comment (bytewiz) [inline src/client/context/userContext.js:15]: Remove `console.log()`

**Review by bytewiz — CHANGES_REQUESTED** [review]
Seems correct, but it only wraps one route. Usually, you would wrap the entire app in the context provider so you can access the context in the entire app. At least that is usually the case for Auth-related context as we need to check it etc.

**Review by yukafjaponesa — COMMENTED** [review]
when bytewiz's feedback implemented, then looks good.

File: src/client/context/UserContextTest.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  ) : (
+    <button onClick={login} type="button">
+      login
+    </button>
+  );
+};
+
+const AppWithStore = () => {
+  return (
+    <UserProvider>
```
Comment (tpetrina) [inline src/client/context/UserContextTest.js:19]: You don't need UserProvider here.

File: src/client/context/UserContextTest.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      <TestButton />
+    </UserProvider>
+  );
+};
+
+export default AppWithStore;
```
Comment (tpetrina) [inline src/client/context/UserContextTest.js:25]: You are importing this as `UserTest` in `App.js`, keep names the same.

**Review by tpetrina — CHANGES_REQUESTED** [review]
Please address all comments.

**Review by tpetrina — DISMISSED** [review]
Fix conflict and you are good to go!

**Review by tpetrina — DISMISSED** [review]
Good stuff!

**Review by Pallabi-ghosh — DISMISSED** [review]

**Review by Yaser-Aldammad — DISMISSED** [review]

**Review by tpetrina — DISMISSED** [review]

**Review by Elia-Git — APPROVED** [review]
Good job!

**Review by dpfernandes — APPROVED** [review]
👍

**Pull request merged**